### ISSUE TITLE
Combobox: calls startup() illegally

### ISSUE THREAD

**Issue by wkeese** [ibm-js/deliteful#482]
The rule for calling startup() or attachedCallback() is that the DOM node needs to be attached to the document.

tests/unit/Combobox.js violates this rule by calling startup() here:

```js
var initCombobox = function (combo, trackable) {
	var TrackableMemoryStore = Memory.createSubclass(Trackable);
	combo.list.store = trackable ?
		new TrackableMemoryStore({}) : new Memory();
	var dataItems = addOptions(combo, 0, nOptions - 1);
	combo._testDataItems = dataItems; // stored for debugging purposes.
	combo.startup();
	return combo;
};
```

The obvious fix is to move the startup() call to here:

```js
var createCombobox = function (id, trackable, multiple) {
	var selectionMode = multiple ? "multiple" : "single";
	var combo = new Combobox({ id: id, selectionMode: selectionMode });
	initCombobox(combo, trackable);
	container.appendChild(combo);
	combo.startup();
	return combo;
};
```

However, that causes test failures:

```
Store.add/remove/put (user's trackable Memory store)	Number of options after adding 10 options on combo.id: combo-a-2: expected 0 to equal 10	0:00.025
	Store.add (user's non-trackable Memory store)	Number of options after adding 10 options on combo.id: combo-b-2: expected 0 to equal 10
```

Again, this in on firefox.

I can't really finish the conversion from startup() to attachedCallback() until this and #481 are fixed.

**Comment by AdrianVasiliu** [comment]
Right. For some strange reason, the badly placed startup() in the test was harmless - it doesn't seem to be making the test fail even with the new release of delite (0.6.0-beta) - nor with the previous one. The latest deliteful build (ran after 0.6.0-beta, with a dependency already set on 0.6.x in deliteful) doesn't show any error: 
https://travis-ci.org/ibm-js/deliteful/builds/48391789
How do you reproduce this error? You run with grunt test:local? test:remote? directly in the browser? Using which version of decor/delite?

> The obvious fix is to move the startup() call to here [`createCombobox()`]. However, that causes test failures

I'll not be able to test before this afternoon my time (I'm off this morning), but maybe it's because `initCombobox()` is also called by `createMyCombobox()` (https://github.com/ibm-js/deliteful/blob/master/tests/unit/Combobox.js#L98-103) so `startup()` would also be need there.

**Comment by wkeese** [comment]
> How do you reproduce this error?

I'm not sure what you're asking.  As you said yourself:

> the badly placed startup() in the test was harmless - it doesn't seem to be making the test fail even with the new release of delite (0.6.0-beta) - nor with the previous one - in the travis build.

To be clear, this ticket is for fixing the test to call startup() in the right place, and then resolve the test failure.

See also #481.

**Comment by AdrianVasiliu** [comment]
>> How do you reproduce this error?

> I'm not sure what you're asking. 

Okay, I've read it too fast; you're not getting the test failure with the code as-is, but only once you tentatively fix the badly placed startup() in the test code. I hope fixing similarly `createMyCombobox()` (as I wrote) will do the trick. I'll work on it this afternoon.

**Comment by AdrianVasiliu** [comment]
Pushed the fix via 45b41ca0eea06c253e08ee9ae9eab7352867d22e (moved the call of startup() from initCombobox() to *both* createCombobox() and createMyCombobox()).

In my testing, after this fix, I don't get any test failure on Firefox - tested with FF 30 and 35.0.1 (that is, before and after upgrade to latest available) on both Win7 and Mac OS X 10.9.5, using decor, delite, deliteful freshly updated from master.

**Comment by AdrianVasiliu** [comment]
Just to add that test failures do happen intermitently/randomly (in Combobox test and maybe in other tests) when running locally and somehow disturbing the test by moving the mouse, giving focus etc. To be on the safe side there should be no user interaction while the test runs. (Also - although this is more unlikely - it might be useful to check that browsers zoom is set to 1:1.)
